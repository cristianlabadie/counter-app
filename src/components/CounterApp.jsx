import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value }) => {
  let [counter, setCounter] = useState(value);
  const handleIncrement = () => setCounter(counter + 1);
  const handleDecrement = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2 data-testid="test-counter">{counter}</h2>

      <button onClick={handleIncrement}>+</button>
      <button onClick={handleDecrement}>-</button>
      <button aria-label="btn-reset" onClick={handleReset}>
        Reset
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
