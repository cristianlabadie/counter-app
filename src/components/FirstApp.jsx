import PropTypes from "prop-types";

export const FirstApp = ({ title, subtitle, name }) => {
  return (
    <>
      <h2 data-testid="test-title"> {title} </h2>
      <p data-testid="test-subtitle">{subtitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  name: PropTypes.string,
};

FirstApp.defaultProps = {
  subtitle: "Soy un subtitulo",
};
