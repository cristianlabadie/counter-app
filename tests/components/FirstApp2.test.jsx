import { render, screen } from "@testing-library/react";
import { FirstApp } from "../../src/components/FirstApp";

describe("Pruebas en <FirstApp /> 2", () => {
  const title = "Hola, soy goku";
  const subtitle = "Soy un subtitulo";

  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container.innerHTML).toMatchSnapshot();
  });

  test('debe de mostrar el mensaje "Hola, soy goku', () => {
    render(<FirstApp title={title} />);
    // screen.debug();
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("debe de mostrar el titulo en un h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(
      title,
    );
  });

  test("debe de mostrar el subtitulo enviado por props", () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getByTestId("test-subtitle").innerHTML).toContain(subtitle);
  });
});
