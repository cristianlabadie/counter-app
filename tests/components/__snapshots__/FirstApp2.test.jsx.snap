// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> 2 debe de hacer match con el snapshot 1`] = `"<h2 data-testid="test-title"> Hola, soy goku </h2><p data-testid="test-subtitle">Soy un subtitulo</p><p></p>"`;
